lib/instances/HttpService.lua

```diff
@@ -18,15 +27,25 @@ function HttpService.prototype:GenerateGUID(wrapInCurlyBraces)
 		error(("Unable to cast %s to bool"):format(argType), 2)
 	end
 
+	local zeroPadded = ("%032x"):format(guidCounter)
+	local guid = ""
+	for index = 1, 32 do
+		guid = guid .. zeroPadded:sub(index, index)
+		if GUID_DASH_POINTS[index] then
+			guid = guid .. "-"
+		end
+	end
+	guidCounter = guidCounter + 1

```

Comment:
If we're going to bother generating unique GUIDs, we should use a real GUID generation strategy instead of one that's bound to have collisions across runs!

If someone didn't care about global/universal uniqueness, they would just use an incrementing counter.

Reply:
My rationale was that it's unlikely there'll be a collision with just counting up. Although implementing a real GUID is possible, I don't want to add a new dependency. Can we keep this solution as an alternative if the dependency does not exist?

Reply:
There's no reason to pull in a dependency for generating a GUID. Generating a v4 UUID is trivial with only random number input.

Reply:
Another note is that collisions are definite, because across two runs of the same program, you'll _always_ have a collision with the first GUID you generate.

Reply:
:+1:

Reply:
Resolved.